restli-server/src/test/java/com/linkedin/restli/server/resources/TestInjectResourceFactory.java

```diff
@@ -217,4 +218,36 @@ public void testInjectConstructorArgs()
     assertNotNull(r1.getDependency2());
     assertNull(r1.getNonInjectedDependency());
   }
+
+  @Test
+  public void testInjectDefaultConstructorArgs()
```

Comment:
Seems like this case is exactly the same as the one above. Could you please refactor into one method that accepts the resource class as a parameter + add a data provider to provide `[ConstructorArgResource.class, DefaultConstructorArgResource.class]` to the test method?

Reply:
Could be worthwhile to also test private and protected constructors as well.

Reply:
I suspect that is very uncommonly used and the change itself is covered with the common case of using default protection. Open to both though.

Reply:
There is some argument to be made that protected and private really shouldn't be encouraged, so not having them explicitly tested could discourage it.